### ISSUE TITLE
Question: attributions preprocessing and `get_wrapped_model` only accessible when computing attributions in metric

### ISSUE THREAD

**Issue by vedal** [understandable-machine-intelligence-lab/Quantus#186]
Hi! 
Thanks alot again for this inpiring package!

### Description of the problem
I was trying to pre-compute softmax-attributions `quantus.explain` and evaluate them on a list of metrics through `quantus.evaluate`, and was confused by the design of `quantus` here. 

The main confusion is that: while Quantus gives you the option to evaluate pre-computed attributions on multiple metrics by passing an `a_batch` argument, there is no straight-forward to pre-compute softmax-attributions (cannot be done using `quantus.explain`). Instead, there are two ways with Quantus:

1. use `quantus.evaluate(..., model=model a_batch=None, softmax=True)`, which re-computes softmax-attributions for each metric.
2. leverage `quantus.helpers.utils.get_wrapped_model()` and compute the softmax-attributions manually, as is done in the metric base-class:[base.py:L384](https://github.com/understandable-machine-intelligence-lab/Quantus/blob/8c5a2379f6bc0c8ae30c65b7c4ce19fc967cb0da/quantus/metrics/base.py#L384)) in the following way:
```
a_batch = quantus.explain(model=quantus.helpers.utils.get_wrapped_model(model, softmax=True), x_batch, y_batch)

```

Both of these options are a bit unintuitive IMO, particularly since even the softmax-argument can be hard to spot as a metric-argument. I'd expect it as an argument for both `quantus.explain` or `quantus.evaluate`.

### Description of a solution

- one solution would be to have the softmax-argument accessible from `quantus.explain`? 
- also, for simplicity, maybe `quantus.evaluate` should not allow `a_batch=None` ? 
--------------

More general suggestion:
- attribution preprocessing before metrics should be available as separate modules which could be pipelined, like torchvision.transform or sklearn Pipeline. At the moment, attribution preprocessing, such as `normalise` or `interpolation`, and even attribution computation like softmax, are currently performed under-the-hood in each metric. 

I believe having a separate preprocessing-class, would make the computations of the library more transparent, such that the user can visualize and inspect the final attributions that into the metric.

-----------------------------

Note: a softmax-attributions feature may or may not be a necessary. I see that there are different recommendations mentioned for this, as in this thread: [comment 1](https://github.com/pytorch/captum/issues/407#issue-639582944) and [comment 2](https://github.com/pytorch/captum/issues/407#issuecomment-650684371). I see that captum, for example, only use softmax for [visualization](https://captum.ai/api/insights.html#captum.insights.AttributionVisualizer).
